pkg/templates/antiaffinity/template.go

```diff
@@ -59,30 +61,47 @@ func init() {
 				if !hasPods {
 					return nil
 				}
+				var foundIssues []diagnostic.Diagnostic
 				if affinity := podTemplateSpec.Spec.Affinity; affinity != nil && affinity.PodAntiAffinity != nil {
 					preferredAffinity := affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
 					requiredAffinity := affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution
 					for _, preferred := range preferredAffinity {
-						if affinityTermMatchesLabelsAgainstNodes(preferred.PodAffinityTerm, podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher) {
+						err := affinityTermMatchesLabelsAgainstNodes(preferred.PodAffinityTerm,
+							podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher)
+						if err == nil {
 							return nil
 						}
+						foundIssues = append(foundIssues, diagnostic.Diagnostic{
+							Message: err.Error(),
+						})
 					}
 					for _, required := range requiredAffinity {
-						if affinityTermMatchesLabelsAgainstNodes(required, podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher) {
+						err := affinityTermMatchesLabelsAgainstNodes(required, podTemplateSpec.Namespace,
+							podTemplateSpec.Labels, topologyKeyMatcher)
+						if err == nil {
```

Comment:
Should we do a short circuit here? What if we have an error in next iteration?

Reply:
I was wondering the same thing, I've kept the logic as before, short-circuiting on the first found, valid anti-affinity rule.
I'm fine with changing it to _not_ short-circuit, if we are fine with it.